[bitnami/suitecrm] fix: SKIP_BOOTSTRAP not working properly with suitecrm 8

DavidMarquezF opened the pull request:
<!--
 Before you open the request please review the following guidelines and tips to help it be more easily integrated:

 - Describe the scope of your change - i.e. what the change does.
 - Describe any known limitations with your change.
 - Please run any tests or examples that can exercise your modified code.

 Thank you for contributing! We will try to test and integrate the change as soon as we can, but be aware we have many GitHub repositories to manage and can't immediately respond to every request. There is no need to bump or check in on a pull request (it will clutter the discussion of the request).

 Also don't be worried if the request is closed or not integrated sometimes the priorities of Bitnami might not match the priorities of the pull request. Don't fret, the open source community thrives on forks and GitHub makes it easy to keep your changes in a forked repo.
 -->

### Description of the change

This is my go at making SuiteCRM 8 work properly with `SKIP_BOOSTRAP` flag set to true, which currently doesn't work. I think when migrating from 7 to 8 this path was missed.

This flag allows you to not have to keep the whole SuiteCRM installation in a persisted volume by initializing everything from scratch but the database. 

The thing that still doesn't work which I haven't had a go at is the SMTP set up. 

What I've solved:

* Create .env.local file to support new SuiteCRM 8 config, as well as added necessary properties and updates to the legacy config.php (like cron, and unique_key)
* .htacces file at public/legacy/.htaccess was included, which blocked necessary resources to run the page


This has been more of a surgical fix, but in general the code should/could be cleaned up a lot to remove all the SuiteCRM 7 related code since it's not officially supported anymore by bitnami


### Benefits

`SKIP_BOOTSTRAP` flag functional again.

### Possible drawbacks

* Haven't had time to test the other ways (without SKIP_BOOTSTRAP), but in general my changes should not affect.
* Not really sure why `public/legacy/.htaccess` was included in the first place, but it was also something that did not work properly when running SuiteCRM 7, as it blocks resoruces like images and icons behind 403. Please let me know if this is maybe introducing any security holes

### Applicable issues

<!-- Enter any applicable Issues here (You can reference an issue using #) -->


### Additional information

<!-- If there's anything else that's important and relevant to your pull request, mention that information here.-->

DavidMarquezF committed:
fix(suitecrm): SKIP_BOOTSTRAP not working properly with suitecrm 8

* Create .env.local file to support new SuiteCRM 8 config, as well as added necessary properties and updates to the legacy config.phpo
* .htacces file at public/legacy/.htaccess was included, which blocked necessary resources.

Signed-off-by: David Márquez Fàbrega <32216320+DavidMarquezF@users.noreply.github.com>

javsalgar commented:
Hi!

Thank you so much for the PR! Could you fix the DCO issue?

DavidMarquezF commented:
.

DavidMarquezF commented:
Nice way to accept contributions

migruiz4 reviewed (commented):
Thank you very much for your contribution @DavidMarquezF!

I'm sorry for the late response, could you please take a look at my comments?

DavidMarquezF committed:
fix(): Review changes

Signed-off-by: David Márquez Fàbrega <32216320+DavidMarquezF@users.noreply.github.com>

DavidMarquezF committed:
feat(HTTP_REFERERS):  Added support for http referers list

This is necessary when running SuiteCRM with ACA for instance. The host name is the ACA hostname but the real hostname needs to be enabled here in order to not get blocked for XSRF for some queries
